packages/design-system/src/components/pill/pillGroup.tsx

```diff
@@ -33,7 +33,19 @@ const Container = styled.div.attrs({
   column-gap: 4px;
 `;
 
-function PillGroup({ options, value, onSelect, ...rest }) {
+interface PillOption {
+  id: string;
+  label: string;
+}
+
+interface PillGroupProps
+  extends Omit<ComponentPropsWithoutRef<'button'>, 'onSelect' | 'onClick'> {
+  options: PillOption[];
+  value: string;
+  onSelect: (id: string) => void;
```

Comment:
No need to add `onClick` back?

Reply:
We manually assign an `onClick` handler to the element in question and we don't want a consumer to override that by specifying their own `onClick`, so we just don't allow one to be passed in.